gammapy/data/data_store.py

```diff
@@ -662,12 +662,16 @@ def make_psf(self, position, energy=None, theta=None):
 
         offset = position.separation(self.pointing_radec)
         if not energy:
-            energy = self.psf.to_table_psf(theta=offset).energy
+            energy = self.psf.to_energy_dependent_table_psf(theta=offset).energy
         if not theta:
-            theta = self.psf.to_table_psf(theta=offset).offset
+            theta = self.psf.to_energy_dependent_table_psf(theta=offset).offset
 
-        psf_value = self.psf.to_table_psf(theta=offset, offset=theta)\
-            .evaluate(energy)
+        if (type(self.psf) == PSF3D):
```

Comment:
Please put:
```
if isinstance(self.psf, PSF3D)
```
It's basically the same thing (difference is that it also works for sub-classes).
But it's the more common thing to do, use `isinstance` instead of `type` for type checking, so it's mainly for consistency with other parts of the code.